### PR TITLE
add & update several sql statements

### PR DESCRIPTION
**Pull request by bb7133**
### What is changed, added or deleted? (Required)
add & update several sql statements:

 - sql-statements/sql-statement-show-builtins.md
 - sql-statements/sql-statement-show-engines.md
 - sql-statements/sql-statement-show-builtins.md

### Which TiDB version(s) do your changes apply to? (Required)

- [ ] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: sql-statements/sql-statement-show-builtins.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+| xor                         |
+| year                        |
+| yearweek                    |
++-----------------------------+
+268 rows in set (0.00 sec)
+```
+
+## MySQL 兼容性
+
+`SHOW BUILTINS` 语句功能与 MySQL 完全兼容。但尚有一些极少被用到的函数未被支持，若对这些尚未支持的函数有需求，请在 GitHub 上提交 [issue](/report-issue.md)。
```
Comment (SunRunAway) [inline sql-statements/sql-statement-show-builtins.md:304]: 这里没看懂。 MySQL 有 `SHOW BUILTINS` 语句吗？

File: sql-statements/sql-statement-show-builtins.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
+---
+title: SHOW BUILTINS
+summary: TiDB 数据库中 SHOW BUILTINS 的使用概况。
+category: reference
+aliases: ['/docs-cn/dev/reference/sql/statements/show-builtins/']
```
Comment (yikeke) [inline sql-statements/sql-statement-show-builtins.md:5]:
```suggestion
```
新增的文件不需要加 alias，aliases 属性是用来做 url 重定向的

File: sql-statements/sql-statement-show-builtins.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+---
+
+# SHOW BUILTINS
+
+`SHOW BUILTINS` 语句列出 TiDB 中所有的内置函数。
```
Comment (yikeke) [inline sql-statements/sql-statement-show-builtins.md:10]:
```suggestion
`SHOW BUILTINS` 语句用于列出 TiDB 中所有的内置函数。
```

File: sql-statements/sql-statement-show-engines.md [hunk 4]
```diff
@@ -7,10 +7,14 @@ aliases: ['/docs-cn/dev/reference/sql/statements/show-engines/']
 
 # SHOW ENGINES
 
-`SHOW ENGINES` 语句仅提供 MySQL 兼容性。
+`SHOW ENGINES` 语句列出所有支持的存储引擎，该语法仅提供 MySQL 兼容性。
```
Comment (yikeke) [inline sql-statements/sql-statement-show-engines.md:10]:
```suggestion
`SHOW ENGINES` 语句用于列出所有支持的存储引擎，该语法仅提供 MySQL 兼容性。
```

File: sql-statements/sql-statement-show-builtins.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
+---
```
Comment (yikeke) [inline sql-statements/sql-statement-show-builtins.md:1]: Please add this file link to TOC.md

File: sql-statements/sql-statement-show-builtins.md (hunk 1 above)
Comment (bb7133) [reply sql-statements/sql-statement-show-builtins.md:304]: 原来 MySQL 没有这个。。。

Comment (bb7133) [reply sql-statements/sql-statement-show-builtins.md:304]: 用中文感觉真爽啊，打开了新世界一样

**Comment by bb7133** [comment]
Addressed, PTAL @yikeke @SunRunAway

**Pull request merged**